engines/ospray/ispc/geometry/RayMarching.isph

```diff
@@ -0,0 +1,173 @@
+/* Copyright (c) 2015-2018, EPFL/Blue Brain Project
+ * All rights reserved. Do not distribute without permission.
+ * Responsible Author: Jonas Karlsson <jonas.karlsson@epfl.ch>
+ *
+ * This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+// ospray
+#include "ospray/SDK/common/Ray.ih"
+#include "ospray/SDK/geometry/Geometry.ih"
+#include "ospray/SDK/math/box.ih"
+#include "ospray/SDK/math/vec.ih"
+
+/////////////////////////////////////////////////////////////////////////////
+
+#define SDF_EPSILON 0.0001
+
+/////////////////////////////////////////////////////////////////////////////
+
+// uniform void pointer
+typedef const uniform uint64* uniform dataPtr_t;
+// function pointer type for SDF
+typedef float (*distanceFunction_t)(const vec3f& /*p*/, dataPtr_t /*geo*/,
+                                    dataPtr_t /*prim*/,
+                                    const vec4f& /*params*/);
+// function pointer type for SDF aabb compute
+typedef uniform box3fa (*bboxFunction_t)(dataPtr_t /*geo*/, dataPtr_t /*prim*/);
+
+/////////////////////////////////////////////////////////////////////////////
+
+inline vec3f computeNormal(const vec3f& pos,
+                           const uniform distanceFunction_t sdfDistance,
+                           const float e, dataPtr_t geo, dataPtr_t prim,
+                           const vec4f& sdfParams)
+{
+// #define USE_CENTRAL_DIFFERENCES
+#ifdef USE_CENTRAL_DIFFERENCES
```

Comment:
Is there any visible difference between these two methods? If not we can remove the old version.